File: common/src/main/java/com/wolfyscript/utilities/gui/animation/ButtonAnimationFrameImpl.java
package com.wolfyscript.utilities.gui.animation;

import com.wolfyscript.utilities.gui.GuiHolder;
import com.wolfyscript.utilities.gui.ViewRuntime;
import com.wolfyscript.utilities.gui.rendering.RenderContext;
import com.wolfyscript.utilities.world.items.ItemStackConfig;

public class ButtonAnimationFrameImpl implements ButtonAnimationFrame {

    private final Animation<ButtonAnimationFrame> animation;
    private final ItemStackConfig stack;
    private final int duration;

    protected ButtonAnimationFrameImpl(Animation<ButtonAnimationFrame> animation, int duration, ItemStackConfig stack) {
        this.animation = animation;
        this.duration = duration;
        this.stack = stack;
    }

    @Override
    public int duration() {
        return duration;
    }

    @Override
    public Animation<ButtonAnimationFrame> animation() {
        return animation;
    }

    @Override
    public void render(ViewRuntime viewManager, GuiHolder holder, RenderContext context) {

    }

    @Override
    public ItemStackConfig stack() {
        return stack;
    }

}

File: common/src/main/java/com/wolfyscript/viewportl/common/gui/animation/ButtonAnimationFrameImpl.kt
/*
 *     viewportl - multiplatform GUI framework to easily create reactive GUIs
 *     Copyright (C) 2024  WolfyScript
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation, either version 3 of the License, or
 *     (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
package com.wolfyscript.viewportl.common.gui.animation

import com.wolfyscript.scafall.wrappers.world.items.ItemStackConfig
import com.wolfyscript.viewportl.gui.GuiHolder
import com.wolfyscript.viewportl.gui.ViewRuntime
import com.wolfyscript.viewportl.gui.animation.Animation
import com.wolfyscript.viewportl.gui.animation.ButtonAnimationFrame
import com.wolfyscript.viewportl.gui.rendering.RenderContext

class ButtonAnimationFrameImpl(
    private val animation: Animation<ButtonAnimationFrame>,
    private val duration: Int,
    private val stack: ItemStackConfig
) :
    ButtonAnimationFrame {
    override fun duration(): Int {
        return duration
    }

    override fun animation(): Animation<ButtonAnimationFrame> {
        return animation
    }

    override fun render(viewManager: ViewRuntime<*,*>, holder: GuiHolder, context: RenderContext) {
    }

    override fun stack(): ItemStackConfig {
        return stack
    }
}
